cnd/src/swap_protocols/swap_communication_state.rs

```diff
@@ -0,0 +1,36 @@
+use crate::swap_protocols::{
+    state::{Get, Insert},
+    swap_id::SwapId,
+};
+use std::{any::Any, clone::Clone, collections::HashMap, sync::Mutex};
+
+#[derive(Default, Debug)]
+pub struct SwapCommunicationState {
```

Comment:
WDYT of naming these in plural because they contain all of them?

```suggestion
pub struct SwapCommunicationStates {
```

Reply:
Makes more sense. All these new names are a bit odd still.

Reply:
Pretty sure we will figure something out as we iterate :)